### ISSUE TITLE
Audio Data Streaming too slow using http2 post through libcurl

### ISSUE THREAD

**Issue by Nitya05** [curl/curl#2961]
I am trying to send some audio data to a server using http post (curl enabled with nghttp2). The audio data is being recorded by a microphone and is sent in buffers and the sending process is stopped only after the server confirms that it has read a complete sentence (that is, it detects silence after receiving one full statement).
But the data sending process is so slow that the server times out after receiving 2 or 3 packets. The problem is that the server is expecting packets at a speed of around 2 packets in a second, else it will timeout, and currently it is taking around 600-700 ms for a single packet [Note: I cannot make any changes on the server side]. I even tried using multicurl to speed up the process, enabling the multiplexing option using CURLMOPT_PIPELINING; still, there was no improvement. This option creates a new connection with the server, everytime it is trying to send a new packet, which in turn takes even more time. I could stop the creation of multiple connections by making CURLMOPT_MAX_TOTAL_CONNECTIONS to 1, but still the packet sending time is not optimal.

Is there anything that I can do to reduce this time, may be by sending all the packets in parallel, but keeping their order intact?

### curl/libcurl version
libcurl version: 7.50.2
nghttp2 version: 1.0.0
openssl version: 1.0.2a

### operating system
Ubuntu 16.04

**Comment by bagder** [comment]
First: you're leaving out a whole lot of information, that leaves us guessing. For example, how far away RTT wise is the server and how large is the content you post == how long *should* this take in the ideal case compared to the *slow* case you see. You also don't tell us much about what your application does with libcurl like options used etc. It would help us if you perhaps modified an example code to reproduce your problem. I presume you use HTTPS since you mention openssl? Are you doing more than one POST in parallel to this single server?

Then: we've fixed many (http2) bugs in curl since 7.50.2. The one I suspect has the most impact for you might be the [upload fix](https://curl.haxx.se/bug/?i=2520) we did in 7.60.0. But we've done more fixes since too and 7.61.1 has a whole bunch of them. I recommend at least trying with the latest version first to see if the problems are gone or remain.

Your nghttp2 version is also very old and we know they fixed problems that could crash libcurl as recently as in 1.33.0, released just days ago. I would also recommend that you at least try with the latest there before you spend a lot of time chasing this problem down. It might just be fixed already.
